consensus/types/src/chain_spec.rs

```diff
@@ -1833,6 +1720,18 @@ impl Config {
         }
     }
 
+    /// As discussed previously, sometimes configs contain extra fields for future forks
+    /// This is intended to check those extra fields to see if it's for a future fork or not
+    pub fn validate_extra_fields(&self) -> Result<(), String> {
+        for key in self.future_fields.keys() {
+            if !FUTURE_FIELDS.contains(&key.as_str()) {
+                return Err(format!("Unknown or disallowed field found: {}", key));
+            }
+        }
+
+        Ok(())
+    }
```

Comment:
I like this approach. Simple and flexible.

Let me know if you get stuck fixing any of the tests